library/test/src/lib.rs

```diff
@@ -588,6 +589,17 @@ fn run_test_in_process(
     monitor_ch: Sender<CompletedTest>,
     time_opts: Option<time::TestTimeOptions>,
 ) {
+    // If we encounter a non-unwinding panic, stop capturing output to ensure that the
+    // non-unwinding panic message is visible.
+    let builtin_panic_hook = panic::take_hook();
+    let hook = Box::new(move |info: &'_ PanicInfo<'_>| {
+        if !info.can_unwind() {
+            io::set_output_capture(None);
```

Comment:
There should also be a method for stderr. Additionally it would be nice to have some code to write out the already captured content.

Reply:
Is there a method for stderr? It looks to me like both stdout and stderr are sent to the single output capture buffer.

Reply:
Looks like the stdout and stderr methods got merged in https://github.com/rust-lang/rust/commit/aff7bd66e8b97a41d34c221007e12e4bbe535322.